### PR TITLE
FE | `SessionsController#omniauth` 

### PR DESCRIPTION
**Pull request by KaraJoHo**
# Changes 
[ ] Bug fix
[ x] New feature
[ x] This change requires a documentation update

## List of Changes: 
- <p>This PR is related to this issue: #6 </p>
- Adds sessions controller for user login through google
- Refactors sessions controller with sessions facade
- Adds omniauth mock for google oauth testing
- Currently redirects back to login page(will redirect to homepage eventually)
- Create user method for the good deed service, to create the user's session
- Happy path testing completed. sad path tests to be determined
- Syntax and spacing fixes for rubocop

# Self Review
[x ] I have performed a self-review of my code.
[x ] I have added thorough tests.
[x ] My changes generate no new warnings.
[ x] I have linted my code locally before submission.

### CODE REVIEW

**Review by hamouj — APPROVED** [review]
Noted the changes for styling based on Rubocop and the inclusion of omniauth to do Google OAuth as well as stubbing of the OAuth request.

Everything is passing--nice job!

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# app/controllers/sessions_controller.rb
+class SessionsController < ApplicationController
+  def omniauth
+    user = SessionsFacade.new.authorize_user(request.env["omniauth.auth"])
+    session[:user_id] = user.name
```
Comment (MelTravelz) [inline app/controllers/sessions_controller.rb:7]: Brilliant idea to save the user name so we don't need to ask BE for that data at any other point throughout a session!

File: app/services/do_good_service.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# app/services/do_good_service.rb
+class DoGoodService
+  def self.create_user(oauth_hash)
+    response = conn.post("/api/v1/users") do |req|
+      req.body = { query: oauth_hash }.to_json
```
Comment (MelTravelz) [inline app/services/do_good_service.rb:7]: This is so very interesting! Thanks for explaining!!

**Review by MelTravelz — APPROVED** [review]
Absolutely amazing work team. This looks great!

**Pull request merged**